app/src/main/java/tech/pauly/findapet/dependencyinjection/ApplicationComponent.java

```diff
@@ -5,12 +5,13 @@
 import dagger.Component;
 import dagger.android.AndroidInjector;
 import dagger.android.support.AndroidSupportInjectionModule;
+import tech.pauly.findapet.shared.PermissionHelper;
 
 @Singleton
 @Component(modules = { AndroidSupportInjectionModule.class,
                        AndroidViewModule.class,
                        ApplicationModule.class,
                        DataModule.class })
 public interface ApplicationComponent extends AndroidInjector<PetApplication> {
-
+    PermissionHelper permissionHelper();
```

Comment:
This is an unconventional way of using Dagger in my experience, but I've found it useful for one specific use case:

You have a dependency in the Dagger object graph that a class that cannot itself be in the object graph needs.

In this case, the PermissionHelper is a singleton in the Dagger graph, and the BaseActivity needs it. We could just use Dagger-Android to inject it into the BaseActivity, but then all activities will require themselves to be registered in the AndroidViewModule. I don't like this requirement especially as it can sneak up on you, so I'd prefer the BaseActivity to be able to statically import this class into itself. This obviously breaks unit testing, but we don't unit test Activities anyway so 👍